File: FoodChainWebShop/Models/Restaurant.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace FoodChainWebShop.Models {
    public class Restaurant {
        public int RestaurantId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string imageName { get; set; }

        [Required]
        [Phone (ErrorMessage = "Entered number is invalid!")]
        public string mobileNumber { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public int minOrderPrice { get; set; }

        [Required]
        public int UserId { get; set; }

        [JsonIgnore]
        public User User { get; set; }

        [JsonIgnore]
        public ICollection<Product> Products { get; set; }

        [Required]
        public int RestaurantTypeId { get; set; }
        public RestaurantType RestaurantType { get; set; }

    }
}

File: FoodChainWebShop/ClientApp/src/app/interfaces/Restaurant.ts
import { SafeResourceUrl } from "@angular/platform-browser";

export interface Restaurant {
    restaurantId: number, 
    restaurantTypeId: number, 
    userId: number
    name: string,
    image?: ArrayBuffer | File | string,
    imageToShow?: SafeResourceUrl,
    mobileNumber: string,
    address: string,
    minOrderPrice: number,
    RestaurantTypeId: number,
    active: boolean,
    imageFile?: any,
}